src/verify.c

```diff
@@ -474,6 +478,30 @@ static void add_missing_files(struct manifest *official_manifest)
 		if ((ret != 0) || hash_needs_work(file, local.hash)) {
 			file_not_replaced_count++;
 			printf("\n\tnot fixed\n");
+
+			if (freespace_warning_flag && cmdline_option_install) {
+				fs_free = get_available_space(path_prefix);
+				string_or_die(&original, "%s/staged/%s", state_dir, file->hash);
+				if (stat(original, &st) == 0) {
+					if (((st.st_size * 1.1) > fs_free) || fs_free < 0) {
+						if (fs_free >= 0) {
+							fprintf(stderr, "\tWarning: File to install (%s) too large by %ldK.\n", file->filename, (st.st_size - fs_free) / 1000);
+							no_freespace_flag = true;
+							/* no longer a warning - we are confirmed out of free disk space. */
+							freespace_warning_flag = false;
+						} else if (fs_free < 0) {
+							fprintf(stderr, "\tWarning: Unable to determine free space on filesystem.\n");
+						}
+						fprintf(stderr, "\tContinuing with install...\n");
+					}
+				} else {
+					fprintf(stderr, "\tWarning: Unable to determine file size for staged file %s", original);
+				}
+				free_string(&original);
+			}
+			if (no_freespace_flag) {
+				fprintf(stderr, "\tWarning: No space left on device\n");
```

Comment:
I I understood the code correctly, this message will be printed only once. You can move this print go line 489 and get rid of the no_freespace_flag value.

Reply:
This message is printed once per file that fails (after the first file fails due to space constraints). The flag is there so we don't run a space check on each failed file - decreasing the time it takes to complete the command.